tests/Checker/Check_Result_Tests.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * Tests for the Check_Result class.
+ *
+ * @package plugin-check
+ */
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+class Check_Result_Tests extends WP_UnitTestCase {
+	public function set_up() {
+		parent::set_up();
+
+		$check_context = new Check_Context( 'test-plugin/test-plugin.php' );
+
+		$this->check_result = new Check_Result( $check_context );
+	}
+
+	public function test_plugin() {
+		$this->assertInstanceOf( Check_Context::class, $this->check_result->plugin() );
```

Comment:
Maybe also check that the context has the expected basename based on what is passed in the `set_up` method?

Reply:
Thanks @felixarntz I've added an additional assertion for the `Check_Context` `basename()`